examples/rust/get_started/src/attribute_access_control.rs

```diff
@@ -0,0 +1,93 @@
+use ockam::abac::Expr::*;
+use ockam::abac::{eval, Env, Expr};
+use ockam::authenticated_storage::AuthenticatedStorage;
+use ockam::identity::credential::AttributesStorageUtils;
+use ockam::identity::IdentitySecureChannelLocalInfo;
+use ockam::Result;
+use ockam_core::async_trait;
+use ockam_core::compat::boxed::Box;
+use ockam_core::{IncomingAccessControl, RelayMessage};
+use std::fmt::{Debug, Formatter};
+use std::str;
+use std::sync::Arc;
+
+/// Create an IncomingAccessControl which will verify that the sender of
+/// a message has an authenticated attribute with the correct name and value
+pub fn create_attribute_access_control<S>(
+    storage: S,
+    attribute_name: &str,
+    attribute_value: &str,
+) -> Arc<dyn IncomingAccessControl>
+where
+    S: AuthenticatedStorage,
+{
+    let expression = List(vec![
+        Ident("=".into()),
+        Ident(format!("subject.{attribute_name}")),
+        Str(attribute_value.into()),
+    ]);
+    Arc::new(AbacAccessControl::new(storage, expression))
+}
+
+/// This AccessControl uses a storage for authenticated attributes in order
+/// to verify if a policy expression is valid
+pub struct AbacAccessControl<S> {
+    attributes_storage: S,
+    expression: Expr,
+}
+
+impl<S> Debug for AbacAccessControl<S> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let expression = self.expression.clone();
+        f.write_str(format!("{expression:?}").as_str())
+    }
+}
+
+impl<S> AbacAccessControl<S> {
+    fn new(attributes_storage: S, expression: Expr) -> Self {
+        Self {
+            attributes_storage,
+            expression,
+        }
+    }
+}
+
+#[async_trait]
+impl<S> IncomingAccessControl for AbacAccessControl<S>
```

Comment:
I wonder what are the difference between this and the original implementation

Reply:
This one is a simpler implementation which does not store the policy

Reply:
What about whitespaces?